src/containers/DAO/pages/DAOPage/Proposals/CreateProposal/hooks/useCreateProposalForm.ts

```diff
@@ -0,0 +1,319 @@
+import { useState, useMemo, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { isEqual } from 'lodash';
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider';
+import type { Proposal, zDAO } from '@zero-tech/zdao-sdk';
+import type { Option } from 'components/Dropdowns/OptionDropdown';
+import { parseUnits } from 'ethers/lib/utils';
+import { useUpdateEffect } from 'lib/hooks/useUpdateEffect';
+import { usePropsState } from 'lib/hooks/usePropsState';
+import { useValidateRouteChanges } from 'lib/hooks/useValidateRouteChanges';
+import { useProposals } from 'lib/dao/providers/ProposalsProvider';
+import config from 'config';
+import {
+	ProposalInputFieldKeys,
+	ProposalFormDefaultValues,
+	ProposalFormDefaultErrors,
+	DEFAULT_VOTE_DURATION_SECONDS,
+	DEFAULT_VOTE_CHOICES,
+	NEW_PROPOSAL_TWEET_OPTION,
+} from '../CreateProposal.constants';
+import {
+	getTokenOption,
+	getVotingDetails,
+	isValidERC20Address,
+	validateCreateProposalForm,
+} from '../CreateProposal.helpers';
+import { DAO_CREATE_PROPPAL } from '../../../Proposals/Proposals.constants';
+
+export const useCreateProposalForm = ({
```

Comment:
This hook is doing too much. Hooks should have a clearly defined purpose - at the moment this is driving the entirety of the UI and handling the business logic.

Can we break this down to be simpler?